### DISCUSSION TITLE
2 connected BT devices and metadata callback

### DISCUSSION THREAD

**Discussion by Shavo650** [pschatzmann/ESP32-A2DP#350] [Q&A] [answered]
Hi all!

So here is the thing. I want to use 2 BT devices in my project:
1. CSR8645 - to send music there
2. ESP32 - to get metadata (artist, title) and send it to LCD display.
So i connected my phone to 2 BT devices at once and it works! Almost ... The problem is that ESP32 doesn't receive any metadata until i switch audio output at my phone from CSR8645 to ESP32. I also added in my code if statement.
` if (a2dp_sink.get_audio_state() == ESP_A2D_AUDIO_STATE_STARTED) {do something}`
But it has same problem as above. Is it any way to fix this other than use ESP32 instead od CSR8645 to stream music ? Maybe there is possibility to request metadata instead than relying on callback

**Comment by pschatzmann** [comment, marked as answer]
I guess you would need to reprogram your phone and my library to make this work.
